### PR TITLE
Upgrade to Dotty 0.20.0-RC1

### PR DESCRIPTION
**Pull request by bishabosha**
will help finish https://github.com/lampepfl/dotty/issues/7479

### CODE REVIEW

**Comment by nicolasstucki** [comment]
This should be rebased on master

File: src/test/scala/dotty/xml/interpolator/ReporterTest.scala [hunk 1]
```diff
@@ -57,83 +57,83 @@ class ReporterTest {
   }
 
   @Test def attribute1(): Unit = {
-    assertEquals(xml"<foo bar", List((8, "'=' expected but end of source found")))
+    assertEquals(xml"<foo bar", List((5, "'>' expected but 'b' found")))
```
Comment (nicolasstucki) [inline src/test/scala/dotty/xml/interpolator/ReporterTest.scala]: Why are the outputs different?

Comment (nicolasstucki) [reply src/test/scala/dotty/xml/interpolator/ReporterTest.scala]: Are the parser combinators behaving differently in `"scala-parser-combinators" % "1.1.2"`?

Comment (nicolasstucki) [reply src/test/scala/dotty/xml/interpolator/ReporterTest.scala]: Or `"scala-xml" % "1.2.0"` changed something?

Comment (bishabosha) [reply src/test/scala/dotty/xml/interpolator/ReporterTest.scala]: It looks like the scala-parser-combinators version change affected the errors?

Comment (nicolasstucki) [reply src/test/scala/dotty/xml/interpolator/ReporterTest.scala]: It looks ok

**Pull request merged**